### PR TITLE
guides: fix nits in buffer-constructor-deprecation

### PR DESCRIPTION
**Pull request by vsemozhetbyt**
This PR unify some formatting differences across the doc and fix some other nits.

### CODE REVIEW

File: locale/en/docs/guides/buffer-constructor-deprecation.md [hunk 1]
```diff
@@ -31,10 +31,11 @@ If you’re using Node.js ≥ 8.0.0 (which is recommended), Node.js exposes mult
...
 - `--pending-deprecation` will show more types of deprecation warnings. In particular, it will show the `Buffer()` deprecation warning, even on Node.js 8.
 
-You can set these flags using an environment variable:
+You can set these flags using environment variables:
 
-```console
-$ export NODE_OPTIONS='--trace-warnings --pending-deprecation'
+```bash
+$ export NODE_OPTIONS=--trace-warnings
+$ export NODE_PENDING_DEPRECATION=1
```
Comment (vsemozhetbyt) [inline locale/en/docs/guides/buffer-constructor-deprecation.md]: It seems [we have not](https://nodejs.org/download/nightly/v11.0.0-nightly20180530c1a87a30e7/docs/api/cli.html#cli_node_options_options) `--pending-deprecation` in `NODE_OPTIONS`.

File: locale/en/docs/guides/buffer-constructor-deprecation.md [hunk 2]
```diff
@@ -31,10 +31,11 @@ If you’re using Node.js ≥ 8.0.0 (which is recommended), Node.js exposes mult
 - `--trace-deprecation` does the same thing, but only for deprecation warnings.
 - `--pending-deprecation` will show more types of deprecation warnings. In particular, it will show the `Buffer()` deprecation warning, even on Node.js 8.
 
-You can set these flags using an environment variable:
+You can set these flags using environment variables:
 
-```console
-$ export NODE_OPTIONS='--trace-warnings --pending-deprecation'
+```bash
```
Comment (vsemozhetbyt) [inline locale/en/docs/guides/buffer-constructor-deprecation.md]: `console` tag seems to produce [weird rendering](https://nodejs.org/en/docs/guides/buffer-constructor-deprecation/#finding-problematic-bits-of-code-using-node-js-8):

**Comment by vsemozhetbyt** [comment]
I am a bit confused by [this fragment](https://github.com/nodejs/nodejs.org/blame/6aaedb203b835d7cb356bb80a0dfb2f00ea975c1/locale/en/docs/guides/buffer-constructor-deprecation.md#L163-L164) in the example code. Are the condition check and error message in some contradiction?

**Comment by vsemozhetbyt** [comment]
cc @nodejs/documentation @nodejs/buffer

File: locale/en/docs/guides/buffer-constructor-deprecation.md (hunk 1 above)
Comment (lpinca) [reply locale/en/docs/guides/buffer-constructor-deprecation.md]: @vsemozhetbyt https://nodejs.org/download/nightly/v11.0.0-nightly20180530c1a87a30e7/docs/api/cli.html#cli_pending_deprecation

**Comment by lpinca** [comment]
> I am a bit confused by this fragment in the example code. Are the condition check and error message in some contradiction?

Yes, I think so the message should be 'The "size" argument must **not** be of type number.'

File: locale/en/docs/guides/buffer-constructor-deprecation.md (hunk 1 above)
Comment (vsemozhetbyt) [reply locale/en/docs/guides/buffer-constructor-deprecation.md]: Yes, we have it as a separate cli key, but not as a `NODE_OPTIONS` env variable part option.

Comment (lpinca) [reply locale/en/docs/guides/buffer-constructor-deprecation.md]: I'm not sure I follow, `--pending-deprecation` is a CLI option right? If so it should work when used via `NODE_OPTIONS` which allows to specify CLI options in the environment via a space-separated list.

Comment (vsemozhetbyt) [reply locale/en/docs/guides/buffer-constructor-deprecation.md]: It seems we have [a whitelist of cli keys that are allowed in `NODE_OPTIONS`](https://nodejs.org/download/nightly/v11.0.0-nightly20180609ef1f130041/docs/api/cli.html#cli_node_options_options):

> Node options that are allowed are: ...

Comment (lpinca) [reply locale/en/docs/guides/buffer-constructor-deprecation.md]: Was it removed recently? It works on Node.js 10.

Comment (lpinca) [reply locale/en/docs/guides/buffer-constructor-deprecation.md]: I think it is simply not listed there.

Comment (vsemozhetbyt) [reply locale/en/docs/guides/buffer-constructor-deprecation.md]: Node.js 10 doc has not it as well:

https://nodejs.org/dist/latest-v10.x/docs/api/cli.html#cli_node_options_options

If it does work, maybe the API docs need updating.

Comment (lpinca) [reply locale/en/docs/guides/buffer-constructor-deprecation.md]: Yes it seems outdated, see https://github.com/nodejs/node/blob/master/src/node.cc#L3156-L3198

Comment (vsemozhetbyt) [reply locale/en/docs/guides/buffer-constructor-deprecation.md]: Thank you. I will revert this change.

**Comment by vsemozhetbyt** [comment]
> Yes, I think so the message should be 'The "size" argument must **not** be of type number.'

@fhinkel Is this an appropriate fix?

File: locale/en/docs/guides/buffer-constructor-deprecation.md (hunk 1 above)
Comment (vsemozhetbyt) [reply locale/en/docs/guides/buffer-constructor-deprecation.md]: PR to update the API doc: https://github.com/nodejs/node/pull/21229

**Comment by fhemberger** [comment]
@vsemozhetbyt Thanks!

**Pull request merged**